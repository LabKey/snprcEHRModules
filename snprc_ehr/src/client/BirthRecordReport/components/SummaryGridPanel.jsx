import React from 'react'
import { Table } from 'react-bootstrap'
import SummaryGridItem from './SummaryGridItem'

export default class SummaryGridPanel extends React.PureComponent {
  render() {
    return (
      <>
        <div className="table-scroll">
          <Table
            bordered condensed hover
            responsive
          >
            <thead>
              <tr>
                <th>Animal Id</th>
                <th>gender</th>
                <th>Birthdate</th>
              </tr>
            </thead>
            <tbody>
              {this.props.summaryData.length === 0 && <tr><td colSpan="3"> No birth records printed </td></tr>}
              {this.props.summaryData.length > 0 && this.props.summaryData.map(row => {
                return (
                  <SummaryGridItem
                    key={ row.Id }
                    row={ row }
                  />
                )
              })}
            </tbody>
          </Table>
        </div>
      </>
    )
  }
}
