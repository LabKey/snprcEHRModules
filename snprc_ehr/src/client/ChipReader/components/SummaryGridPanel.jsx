import React from 'react'
import { Table } from 'react-bootstrap'
import SummaryGridItem from './SummaryGridItem'

export default class SummaryGridPanel extends React.PureComponent {
  render() {
    return (
      <>
        <div className="table-scroll">
          <Table
            bordered condensed hover
            responsive
          >
            <thead>
              <tr>
                <th>Animal Id</th>
                <th>Chip Id</th>
                <th>Date</th>
                <th>temperature</th>
              </tr>
            </thead>
            <tbody>
              { this.props.summaryData.length === 0 && <tr><td colSpan="4"> No animals </td></tr> }
              { this.props.summaryData.length > 0 && this.props.summaryData.map(row => {
                return (
                  <SummaryGridItem
                    key={ row.animalId.value }
                    row={ row }
                  />
                )
              }) }
            </tbody>
          </Table>
        </div>
      </>
    )
  }
}
