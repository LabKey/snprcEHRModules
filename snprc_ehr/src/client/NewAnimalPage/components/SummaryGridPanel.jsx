import React from 'react'
import { Table } from 'react-bootstrap'
import SummaryGridItem from './SummaryGridItem'

export default class SummaryGridPanel extends React.PureComponent {
  render() {
    return (
      <>
        <div className="table-scroll">
          <Table
            bordered condensed hover
            responsive
          >
            <thead>
              <tr>
                <th>Animal Id</th>
                <th>Acquisition Date</th>
                <th>Species</th>
              </tr>
            </thead>
            <tbody>
              {this.props.summaryData.length === 0 && <tr><td colSpan="3"> No new animals </td></tr>}
              {this.props.summaryData.length > 0 && this.props.summaryData.map(row => {
                return (
                  <SummaryGridItem
                    key={ row.id }
                    print={ this.props.print }
                    row={ row }
                  />
                )
              })}
            </tbody>
          </Table>
        </div>
      </>
    )
  }
}
