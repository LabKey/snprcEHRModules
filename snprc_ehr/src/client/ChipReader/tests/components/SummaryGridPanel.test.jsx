/* eslint-disable no-unused-vars */

import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import SummaryGridPanel from '../../components/SummaryGridPanel'
import { summaryData } from '../fixtures/apiTestData'

function flushPromises() {
    return new Promise(resolve => setImmediate(resolve))
}

describe('SummaryGridPanel tests', () => {
    test('Should render empty SummaryGridPanel.', async () => {
        const mathRandomSpy = jest.spyOn(Math, 'random')
        mathRandomSpy.mockImplementation(() => 0.5)

        const wrapper = shallow(<SummaryGridPanel
          summaryData={ summaryData }

        />)
        await flushPromises()
        expect(wrapper).toMatchSnapshot()
        wrapper.unmount()
    })
})
