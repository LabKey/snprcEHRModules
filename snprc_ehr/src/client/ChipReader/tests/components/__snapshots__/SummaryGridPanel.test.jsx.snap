// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryGridPanel tests Should render empty SummaryGridPanel. 1`] = `
<Fragment>
  <div
    className="id-table-scroll"
  >
    <Table
      bordered={true}
      bsClass="table"
      condensed={true}
      hover={true}
      responsive={true}
      striped={false}
    >
      <thead>
        <tr>
          <th>
            Animal Id
          </th>
          <th>
            Chip Id
          </th>
          <th>
            Date
          </th>
          <th>
            temperature
          </th>
          <th>
            location
          </th>
        </tr>
      </thead>
      <tbody>
        <SummaryGridItem
          key="12345.5000 "
          row={
            Object {
              "animalId": Object {
                "location": "2.00-42",
                "time": "2020-08-18T16:47:44.704Z",
                "url": "/labkey/ehr/SNPRC/participantView.view?participantId=12345",
                "value": "12345",
              },
              "chipId": "1C45433F",
              "temperature": "<25.0",
            }
          }
        />
        <SummaryGridItem
          key="11111.5000 "
          row={
            Object {
              "animalId": Object {
                "location": "2.00-42",
                "time": "2020-08-18T16:48:53.150Z",
                "url": "/labkey/ehr/SNPRC/participantView.view?participantId=11111",
                "value": "11111",
              },
              "chipId": "2C45433F",
              "temperature": "26.2",
            }
          }
        />
        <SummaryGridItem
          key="22222.5000 "
          row={
            Object {
              "animalId": Object {
                "location": "2.00-42",
                "time": "2020-08-18T16:49:56.729Z",
                "url": "/labkey/ehr/SNPRC/participantView.view?participantId=22222",
                "value": "22222",
              },
              "chipId": "3C45433F",
              "temperature": "26.3",
            }
          }
        />
      </tbody>
    </Table>
  </div>
</Fragment>
`;
